packages/ui/components/avatar/avatar.test.tsx

```diff
@@ -0,0 +1,199 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+/* eslint-disable playwright/missing-playwright-await */
+import { render, screen, waitFor, act, fireEvent } from "@testing-library/react";
+import { useState } from "react";
+import { beforeAll, vi } from "vitest";
+
+import { Avatar, sizesPropsBySize } from "./Avatar";
+import { AvatarGroup } from "./AvatarGroup";
+
+type sizeKeys = "xxs" | "xs" | "xsm" | "sm" | "md" | "mdLg" | "lg" | "xl";
+
+vi.mock("@radix-ui/react-avatar", async () => {
+  const ImageMock = (props: { [key: string]: string; alt: string }) => {
+    return <img {...props} alt={props.alt} />;
+  };
+  const actual = (await vi.importActual("@radix-ui/react-avatar")) as { default: object };
+  return {
+    ...actual.default,
+    Image: ImageMock,
+  };
+});
+
+const observeMock = vi.fn();
+
+window.ResizeObserver = vi.fn().mockImplementation(() => ({
+  disconnect: vi.fn(),
+  observe: observeMock,
+  unobserve: vi.fn(),
+}));
+
+vi.mock("../tooltip", async () => {
+  const actual = (await vi.importActual("../tooltip")) as any;
+  const TooltipMock = (props: object) => {
+    const [open, setOpen] = useState(false);
+
+    return (
+      <actual.Tooltip
+        {...props}
+        open={open}
+        onOpenChange={(isOpen: boolean) => {
+          setOpen(isOpen);
+        }}
+      />
+    );
+  };
+  return {
+    Tooltip: TooltipMock,
+  };
+});
+
+const restoreRadix = async () => {
+  const mockedLib = (await import("@radix-ui/react-avatar")) as any;
+  const actualLib = (await vi.importActual("@radix-ui/react-avatar")) as any;
+  mockedLib.Image = actualLib.default.Image;
+};
+
+const forcedWaitFor = async () => {
+  await waitFor(
+    async () => {
+      await new Promise((r) => setTimeout(r, 1000));
+    },
+    { timeout: 2000 }
+  );
+};
+
```

Comment:
These two serve to test the fallback, it is necessary to wait for the image to load to fail, for the fallback to appear and that is why the mock of the `Image`, needs to be undone since the default behavior of the radix needs to happen